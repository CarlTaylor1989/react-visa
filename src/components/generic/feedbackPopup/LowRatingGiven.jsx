import React from 'react';
import { Translate } from 'react-redux-i18n';
import PropTypes from 'prop-types';
import CancelButton from './CancelButton';

const LowRatingGiven = ({
  lowRatingFeedback,
  handleChange,
  submitLowRatingFeedback,
  closePopup
}) => (
  <>
    <div className="popupText">
      <Translate value="generic.feedback.popupLowRatingGiven" />
    </div>
    <textarea value={lowRatingFeedback} onChange={handleChange} />
    <div className="buttons">
      <button
        type="button"
        className={`submitBtn${lowRatingFeedback === '' ? ' disabled' : ''}`}
        disabled={lowRatingFeedback === ''}
        onClick={submitLowRatingFeedback}
      >
        <Translate className="text" value="generic.feedback.submitBtn" />
      </button>
      <CancelButton onClick={closePopup} />
    </div>
  </>
);

LowRatingGiven.propTypes = {
  lowRatingFeedback: PropTypes.string.isRequired,
  handleChange: PropTypes.func.isRequired,
  submitLowRatingFeedback: PropTypes.func.isRequired,
  closePopup: PropTypes.func.isRequired
};

export default LowRatingGiven;
