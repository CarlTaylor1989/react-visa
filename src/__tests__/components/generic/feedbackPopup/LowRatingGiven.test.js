/* global toJson */
import React from 'react';
import LowRatingGiven from '../../../../components/generic/feedbackPopup/LowRatingGiven';

let mockProps;
beforeEach(() => {
  jest.resetAllMocks();
  mockProps = {
    lowRatingFeedback: 'Lorem ipsum',
    handleChange: jest.fn(),
    submitLowRatingFeedback: jest.fn(),
    closePopup: jest.fn()
  };
});

test('renders the LowRatingGiven component', () => {
  const wrapper = shallow(<LowRatingGiven {...mockProps} />);
  expect(toJson(wrapper, { noKey: false })).toMatchSnapshot();
});
