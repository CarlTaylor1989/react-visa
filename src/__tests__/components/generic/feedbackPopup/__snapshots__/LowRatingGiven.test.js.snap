// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the LowRatingGiven component 1`] = `
<Fragment>
  <div
    className="popupText"
  >
    <Translate
      tag="span"
      value="generic.feedback.popupLowRatingGiven"
    />
  </div>
  <textarea
    onChange={[MockFunction]}
    value="Lorem ipsum"
  />
  <div
    className="buttons"
  >
    <button
      className="submitBtn"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      <Translate
        className="text"
        tag="span"
        value="generic.feedback.submitBtn"
      />
    </button>
    <CancelButton
      onClick={[MockFunction]}
    />
  </div>
</Fragment>
`;
